### PR TITLE
[REV-1337] Add function that checks an individual against the SDN Fallback

### PR DESCRIPTION
**Pull request by MatthewPiatetsky**
https://openedx.atlassian.net/browse/REV-1337

### CODE REVIEW

File: ecommerce/extensions/payment/core/sdn.py [hunk 1]
```diff
@@ -324,3 +289,63 @@ def populate_sdn_fallback_data_and_metadata(sdn_csv_string):
...
+        1. Order of words doesn’t matter
+        2. Number of times that a given word appears doesn’t matter
+        3. Punctuation between words or at the beginning/end of a given word doesn’t matter
+        4. If a subset of words match, it still counts as a match
+        5. Capitalization doesn’t matter
+    """
+    records = SDNFallbackData.get_current_records_and_filter_by_source_and_type(
+        'Specially Designated Nationals (SDN) - Treasury Department', 'Individual'
+    )
+    records = records.filter(countries__contains=country)
```
Comment (dianekaplan) [inline ecommerce/extensions/payment/core/sdn.py]: Note: openEdX instances will have this code and these tables, but they won't be running the download/import task to fill the tables, so we'll want to make sure we check/handle cases where the SDNFallbackData is empty.

Comment (MatthewPiatetsky) [reply ecommerce/extensions/payment/core/sdn.py]: well right now this function is not hooked up to anything but when we do hook it up I'm assuming the connection will be behind a siteconfiguration

Comment (dianekaplan) [reply ecommerce/extensions/payment/core/sdn.py]: ah, I see- will save that thought for when it's time to review that bit :)

**Review by dianekaplan — APPROVED** [review]

File: ecommerce/extensions/payment/core/tests/test_sdn.py [hunk 2]
```diff
@@ -332,3 +332,85 @@ def test_process_text_unicode(self, text, expected_output):
         """ Verify that characters with accents are transliterated correctly."""
         output = process_text(text)
         self.assertEqual(output, expected_output)
+
+    @ddt.data(
+        # check order properties
```
Comment (jhan217) [inline ecommerce/extensions/payment/core/tests/test_sdn.py:337]: Should we add a case with accented characters?

File: ecommerce/extensions/payment/core/tests/test_sdn.py [hunk 3]
```diff
@@ -332,3 +332,85 @@ def test_process_text_unicode(self, text, expected_output):
...
+        5. Capitalization doesn’t matter
+        """
+        # pylint: disable=line-too-long
+        csv_string = """_id,source,entity_number,type,programs,name,title,addresses,federal_register_notice,start_date,end_date,standard_order,license_requirement,license_policy,call_sign,vessel_type,gross_tonnage,gross_registered_tonnage,vessel_flag,vessel_owner,remarks,source_list_url,alt_names,citizenships,dates_of_birth,nationalities,places_of_birth,source_information_url,ids
+94734218,Specially Designated Nationals (SDN) - Treasury Department,96663868,Individual,material,Juan M. de la Cruz,Dr.,"17472 Christie Stream Apt. 976 North Kristinaport, HI 91033, SN",,,,,,,,,,,,,,https://www.juarez-collier.org/,Wendy Brock,DJ,1944-03-05,Faroe Islands,PK,http://richardson-richardson.org/,CI"""
+        # pylint: enable=line-too-long
+        metadata_entry = populate_sdn_fallback_data_and_metadata(csv_string)
+        self.assertEqual(checkSDNFallback(name, 'North Kristinaport', 'SN'), match)
+
+    @ddt.data(
```
Comment (jhan217) [inline ecommerce/extensions/payment/core/tests/test_sdn.py:387]: Should we add a case with accented characters?

File: ecommerce/extensions/payment/core/tests/test_sdn.py [hunk 4]
```diff
@@ -332,3 +332,85 @@ def test_process_text_unicode(self, text, expected_output):
...
+        """
+        Verify that the following properties are true for addresses:
+        1. Order of words doesn’t matter
+        2. Number of times that a given word appears doesn’t matter
+        3. Punctuation between words or at the beginning/end of a given word doesn’t matter
+        4. If a subset of words match, it still counts as a match
+        5. Capitalization doesn’t matter
+        """
+        # pylint: disable=line-too-long
+        csv_string = """_id,source,entity_number,type,programs,name,title,addresses,federal_register_notice,start_date,end_date,standard_order,license_requirement,license_policy,call_sign,vessel_type,gross_tonnage,gross_registered_tonnage,vessel_flag,vessel_owner,remarks,source_list_url,alt_names,citizenships,dates_of_birth,nationalities,places_of_birth,source_information_url,ids
```
Comment (jhan217) [inline ecommerce/extensions/payment/core/tests/test_sdn.py:415]: Should we include a cases where all the info is the same but one of the following fields don't match?
1. Type is not 'Individual'
2. Country doesn't match
3. City doesn't match

File: ecommerce/extensions/payment/core/tests/test_sdn.py [hunk 5]
```diff
@@ -332,3 +332,85 @@ def test_process_text_unicode(self, text, expected_output):
...
+        """
+        Verify that the following properties are true for names:
+        1. Order of words doesn’t matter
+        2. Number of times that a given word appears doesn’t matter
+        3. Punctuation between words or at the beginning/end of a given word doesn’t matter
+        4. If a subset of words match, it still counts as a match
+        5. Capitalization doesn’t matter
+        """
+        # pylint: disable=line-too-long
+        csv_string = """_id,source,entity_number,type,programs,name,title,addresses,federal_register_notice,start_date,end_date,standard_order,license_requirement,license_policy,call_sign,vessel_type,gross_tonnage,gross_registered_tonnage,vessel_flag,vessel_owner,remarks,source_list_url,alt_names,citizenships,dates_of_birth,nationalities,places_of_birth,source_information_url,ids
```
Comment (jhan217) [inline ecommerce/extensions/payment/core/tests/test_sdn.py:381]: Do we want to test cases against the alt name field?

File: ecommerce/extensions/payment/core/tests/test_sdn.py (hunk 2 above)
Comment (MatthewPiatetsky) [reply ecommerce/extensions/payment/core/tests/test_sdn.py:337]: added

File: ecommerce/extensions/payment/core/tests/test_sdn.py (hunk 3 above)
Comment (MatthewPiatetsky) [reply ecommerce/extensions/payment/core/tests/test_sdn.py:387]: added

File: ecommerce/extensions/payment/core/tests/test_sdn.py (hunk 4 above)
Comment (MatthewPiatetsky) [reply ecommerce/extensions/payment/core/tests/test_sdn.py:415]: added

File: ecommerce/extensions/payment/core/tests/test_sdn.py (hunk 5 above)
Comment (MatthewPiatetsky) [reply ecommerce/extensions/payment/core/tests/test_sdn.py:381]: added

**Review by jhan217 — APPROVED** [review]

**Pull request merged**